### ISSUE TITLE
[Bug report]vuepress unexpected .vue files building

### ISSUE THREAD

**Issue by dousybox** [vuejs/vuepress#1651]
- [x] I confirm that this is an issue rather than a question.

#### Steps to reproduce

```bash
# clone reproduction repo
git clone https://github.com/dousybox/vuepress-unexpected-SFC-building && cd vuepress-unexpected-SFC-building

# install dependencies
npm i

# run docs:build
npm run docs:build
```

#### What is expected?

`Success! Generated static files in docs.`

#### What is actually happening?

Building error with a long error message:

```
too long do not paste
see the reproduce repo
```

In short, it mean that the component can not be found.

#### Other relevant information

- [x] My VuePress version is higher than `1.0.0-alpha.47`, the following is the output of `npx vuepress info` in my VuePress project:

```
Environment Info:

  System:
    OS: macOS 10.14.5
    CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
  Binaries:
    Node: 10.14.2 - /usr/local/bin/node
    Yarn: 1.12.3 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 75.0.3770.80
    Firefox: Not Found
    Safari: 12.1.1
  npmPackages:
    @vuepress/core:  1.0.1 
    @vuepress/theme-default:  1.0.1 
    vuepress: ^1.0.1 => 1.0.1 
  npmGlobalPackages:
    vuepress: Not Found
```

This problem is only repetition in vuepress-1.x.

Every thing is ok in vuepress-0.x, you can checkout the `0.14.11` branch of reproduction repo.

The problem is that vuepress trying to build an irrelevant vue component.
this only repetition in build mode, it's ok in dev mode.

**Comment by dousybox** [comment]
@ulivz 
我在查阅源码后发现这个缺陷是来自于：
https://github.com/vuejs/vuepress/blob/master/packages/%40vuepress/core/lib/node/App.js#L319
此处代码在做模式匹配的时候新增了 `.vue` 文件文件的匹配模式，此模式于 #1601 新增。

这个逻辑导致我在一个Vue前端项目中无法按照～～就近原则～～将文档内容与前端组件源码放置在一起。

至此，有以下两个问题：

Q1：此处能否提供一个参数化配置项来处理匹配模式？

Q2：如果可以，是否可以由你来帮忙完成此项变更？

**Comment by dousybox** [comment]
This bug is fixed in #1705。
Using the `siteConfig.patterns=[**/*.md]` can make the build process work properly.
